libimage/push_test.go

```diff
@@ -96,3 +97,32 @@ func TestPushOtherPlatform(t *testing.T) {
 	_, err = runtime.Push(ctx, "docker.io/library/alpine:latest", "docker-archive:"+tmp.Name(), pushOptions)
 	require.NoError(t, err)
 }
+
+func TestPushWithForceCompression(t *testing.T) {
+	runtime, cleanup := testNewRuntime(t)
+	defer cleanup()
+	ctx := context.Background()
+
+	// Prefetch alpine.
+	pullOptions := &PullOptions{}
+	pullOptions.Writer = os.Stdout
+	pullOptions.Architecture = "arm64"
+	pulledImages, err := runtime.Pull(ctx, "docker.io/library/alpine:latest", config.PullPolicyAlways, pullOptions)
+	require.NoError(t, err)
+	require.Len(t, pulledImages, 1)
+
+	data, err := pulledImages[0].Inspect(ctx, nil)
+	require.NoError(t, err)
+	require.Equal(t, "arm64", data.Architecture)
+
+	pushOptions := &PushOptions{}
+	pushOptions.CompressionFormat = &compression.Gzip
+	pushOptions.ForceCompressionFormat = true
```

Comment:
How could we verify that layers are getting compressed correctly?  Should we load/pull the archive and inspect?

Reply:
Easier and actual integration test would be against a registry, where we can pre-upload a gzip and while pushing with again with `ForceCompressionFormat` we can check push logs that no `reuse` candidates are there, it can be easily verified by inspecting `push logs`.

Reply:
could we push to a local "oci:" directory and inspect the tarball there?

Reply:
let me check that, I'll amend test if that works.

Reply:
I don't think the problem which `ForceCompressionFormat` is solving persists with other transports which works locally, `c/image` force pushes blobs for `oci:` and `dir:` transports when same destination is used even ending up creating invalid OCI images. ( I have added a test and it passes even when `ForceCompressionFormat = false` , I see similar behavior when directly working with buildah )

I am only able to reproduce the problem which `ForceCompressionFormat` with registry as transport where candidates are actually re-evaluated if they can reused before actually `Copying` content. @mtrmac could you confirm this because afaics local transports and not attempting to re-use dest blobs as `registry://` is doing.

Reply:
@mtrmac Thanks :) this worked for me in the test.